toolkit/tools/pkg/imagecustomizerlib/shrinkfilesystems.go

```diff
@@ -69,34 +45,39 @@ func shrinkFilesystems(imageLoopDevice string, verity []imagecustomizerapi.Verit
 		// Check the file system with e2fsck
 		err = shell.ExecuteLive(true /*squashErrors*/, "e2fsck", "-fy", partitionLoopDevice)
 		if err != nil {
-			return fmt.Errorf("failed to check %s with e2fsck:\n%w", partitionLoopDevice, err)
+			return fmt.Errorf("failed to check (%s) with e2fsck:\n%w", partitionLoopDevice, err)
 		}
 
 		// Shrink the file system with resize2fs -M
 		stdout, stderr, err := shell.Execute("flock", "--timeout", "5", imageLoopDevice,
 			"resize2fs", "-M", partitionLoopDevice)
 		if err != nil {
-			return fmt.Errorf("failed to resize %s with resize2fs (and flock):\n%v", partitionLoopDevice, stderr)
+			return fmt.Errorf("failed to resize (%s) with resize2fs (and flock):\n%v", partitionLoopDevice, stderr)
 		}
 
 		// Find the new partition end value
-		end, err := getNewPartitionEndInSectors(stdout, stderr, startSector, imageLoopDevice)
+		filesystemSizeInSectors, err := getFilesystemSizeInSectors(stdout, stderr, imageLoopDevice)
 		if err != nil {
-			return fmt.Errorf("failed to calculate new partition end:\n%w", err)
+			return fmt.Errorf("failed to parse new filesystem size:\n%w", err)
 		}
 
-		if end == "" {
+		if filesystemSizeInSectors < 0 {
 			// Filesystem wasn't resized. So, there is no need to resize the partition.
 			logger.Log.Infof("Filesystem is already at its min size (%s)", partitionLoopDevice)
 			continue
 		}
 
-		// Resize the partition with parted resizepart
-		_, stderr, err = shell.ExecuteWithStdin("yes" /*stdin*/, "flock", "--timeout", "5", imageLoopDevice,
-			"parted", "---pretend-input-tty", imageLoopDevice, "resizepart",
-			strconv.Itoa(partitionNumber), end)
+		// Resize the partition.
+		sfdiskScript := fmt.Sprintf("unit: sectors\nsize=%d", filesystemSizeInSectors)
+
+		err = shell.NewExecBuilder("flock", "--timeout", "5", imageLoopDevice, "sfdisk", "--lock=no",
```

Comment:
are we setting `lock` to `no` because we rely on `flock`?

Reply:
Yeah, `sfdisk` will by default take the lock but it doesn't have a timeout mechanism. So, I use `flock` instead.